tel2puml/otel_to_pv/data_holders/sql_data_holder/sql_dataholder.py

```diff
@@ -230,11 +226,79 @@ def find_unique_graphs(self) -> dict[str, set[str]]:
         """
-        return find_unique_graphs(
-            self.time_buffer,
-            self.batch_size,
-            self
-        )
+        return find_unique_graphs(self.time_buffer, self.batch_size, self)
+
+    def stream_data(
+        self,
+        job_name_to_job_ids_map: dict[str, set[str]] | None = None,
+        filter_job_names: set[str] | None = None,
+    ) -> Generator[tuple[str, Generator[OTelEvent, None, None]], None, None]:
+        """
+        Stream data grouped by job_name from the SQL data holder.
+
+        :param job_name_to_job_ids_map: Optional mapping of job names to job
+        IDs to filter.
+        :type job_name_to_job_ids_map: `dict`[`str`, `set`[`str`]] | `None`
+        :param filter_job_names: Optional set of job names to filter.
+        :type filter_job_names: `set`[`str`] | `None`
+        :return: Generator yielding tuples of (job_name, OTelEvent generator).
+        :rtype: `Generator`[`tuple`[`str`, `Generator`[:class:`OTelEvent`,
+        `None`, `None`]], `None`, `None`]
+        """
+
+        def stream_job_name_batches(
+            session: Session,
+            job_name_to_job_ids_map: dict[str, set[str]] | None = None,
+            filter_job_names: set[str] | None = None,
+        ) -> Generator[tuple[str, OTelEvent], None, None]:
+            """
+            Stream (job_name, OTelEvent) tuples from the database.
+
+            :param session: SQLAlchemy Session object.
+            :type session: :class: `sqlalchemy.orm.Session`
+            :param job_name_to_job_ids_map: Optional mapping of job names to
+            job IDs to filter.
+            :type job_name_to_job_ids_map: `dict`[`str`, `set`[`str`]] | `None`
+            :param filter_job_names: Optional set of job names to filter.
+            :type filter_job_names: `set`[`str`] | `None`
+            :return: Generator yielding (job_name, OTelEvent) tuples.
+            :rtype: `Generator`[`tuple`[`str`, :class: `OTelEvent`], `None`,
+            `None`]
+            """
+            query = session.query(NodeModel)
+            # Apply filters
+            if filter_job_names:
+                query = query.filter(NodeModel.job_name.in_(filter_job_names))
+            if job_name_to_job_ids_map:
+                job_filters = [
+                    (NodeModel.job_name == job_name)
+                    & (NodeModel.job_id.in_(job_ids))
+                    for job_name, job_ids in job_name_to_job_ids_map.items()
+                ]
+                query = query.filter(or_(*job_filters))
+
+            # Order by job_name and job_id to use groupby later on.
+            # Limit query object to batch_size
+            query = query.order_by(
+                NodeModel.job_name, NodeModel.job_id
+            ).yield_per(self.batch_size)
+
+            for node in query:
+                job_name = node.job_name
+                otel_event = self.node_to_otel_event(node)
+                yield job_name, otel_event
```

Comment:
don't need to have job name here as job name is on the otel event anyway and is mandatory